src/actionCreators/index.js

```diff
@@ -113,14 +116,20 @@ export const confirmEmail = (userId, email, jwt) => {
     axi
       .put(API.CONFIRM_SSO_USER(userId), { userEmail: email, token: jwt })
       .then(res => {
+        // Get the updated the current user info after the API request:
+        dispatch(authorize(SELF_SERVER_APP.ROCKETCHAT.NAME, userId));
         return dispatch(confirmEmailSuccess());
       })
       .catch(err => {
-        let errMsg = 'Fail to confirm your email, please register again.';
+        dispatch(authorize(SELF_SERVER_APP.ROCKETCHAT.NAME, userId));
+        // Handle error message based on error code:
+        const hint = '\nPlease register again.';
+        let errMsg = 'Fail to confirm your email.';
         if (err.response) {
-          errMsg = `${err.response.data}. Please register again.`;
+          errMsg = err.response.data;
+          if (err.response.status === 500) errMsg = err.response.data.error;
         }
-        return dispatch(confirmEmailError([errMsg]));
+        return dispatch(confirmEmailError([errMsg.concat(hint)]));
```

Comment:
this will appear to create a 2d array `[ [errMsg, hint] ]`  is that what you are expecting?

Reply:
actually, I'll leave it here for now. Going to refactor all the error messages soon...